src/main/java/dudu/util/Parser.java

```diff
@@ -0,0 +1,75 @@
+package dudu.util;
+
+import dudu.command.*;
+import dudu.exception.DuduException;
+import dudu.exception.EmptyDescriptionException;
+import dudu.task.TaskList;
+
+/**
+ * Parser class on parsing the input.
+ */
+public class Parser {
+    private enum Instruction {
+        TODO,
+        DEADLINE,
+        EVENT,
+        LIST,
+        DELETE,
+        MARK,
+        UNMARK,
+        BYE,
+        HI,
+        FIND
+    }
+
+    /**
+     * Parse command the input.
+     * @param input The user input.
+     * @param list list to store the task.
+     * @param storage store the list
+     * @return Parsed command
+     * @throws DuduException If the command is not recognised.
+     */
+    public static Command parse(String input, TaskList list, Storage storage) throws DuduException {
+        try {
+            String[] inputArr = input.split(" ");
+            Instruction instruction = Instruction.valueOf(inputArr[0].toUpperCase());
+            switch (instruction) {
+            case HI:
+                return new IntroCommand();
+            case TODO:
+                if (input.trim().length() == 4) {
```

Comment:
I like the way you use trim(), you manage to account for the whitespace before and after the command!